crates/common_utils/src/ext_traits.rs

```diff
@@ -434,3 +435,24 @@ impl XmlExt for &str {
         de::from_str(self)
     }
 }
+
+/// Extension trait for logging the error of a Result and then discarding the error
+pub trait ResultExtLog<T> {
+    ///
+    /// Log error if it is of `Err` type else return the `Ok` value wrapped in `Some`
+    ///
+    fn log_err_and_ok(self, key: &str) -> Option<T>;
```

Comment:
```suggestion
    #[track_caller]
    fn log_err_and_ok(self, key: &str) -> Option<T>;
```

Please check if the log message contains the location of where this function has been called, when this annotation is included. If yes, please do include the annotation.

Reply:
I don't think it works that way. But will verify and update here.

Reply:
Yep, there is no difference with and without it. 
with -
![image](https://github.com/juspay/hyperswitch/assets/68317979/7d33ef91-3618-4ac1-81c2-76bbed3e9962)
without -
![image](https://github.com/juspay/hyperswitch/assets/68317979/f37903f6-88d3-4a3b-8b13-219b51be6d88)

Reply:
It was decided to remove this trait based on offline discussions with @vspecky

Reply:
Alright, marking this resolved.